### DISCUSSION TITLE
[Frontend] UI 컴포넌트를 별도 패키지로 관리하기

### DISCUSSION THREAD

**Discussion by rkskekzzz** [80000Coding/80000Coding-Web-Client#86] [General]
## UI 컴포넌트를 별도 패키지로 관리하기

아래는 제가 참고한 예시 레포입니다
https://github.com/vercel/turbo/tree/main/examples/basic

Turborepo + Next.js 로 구성되어 있습니다. 위 예시처럼 저희도 패키지 구성을 수정하면 좋을 것 같습니다
```shell
# 기존
apps
ㄴ /web
    ㄴ /src/component
packages
ㄴ /web-icon

# 변경안
apps
ㄴ /web
packages
ㄴ /web-icon
ㄴ /ui
```

## 장점
위 방법대로 진행했을 때, 제가 생각한 장점입니다.
1. 빌드 속도 개선
nextjs만 에서의 변경이 이뤄진 경우, turborepo에서 build caching이 일어나서 빌드 속도를 엄청나게 향상시킬 수 있습니다.
2. 패키지 관리 및 CI/CD 편리함
storybook과 관련된 모든 기능이 UI 패키지로 분리될 수 있기 때문에, 패키지관리가 쉬워집니다. 추가로 storybook 자동 배포를 만들 때 기존 nextjs 패키지의 스크립트를 건드리지 않고 완전히 독립적으로 구현할 수 있습니다.

<br>

분리해서 개발해도 큰 불편함이 없어보이고, 오히려 장점이 더 크게 다가오는 것 같습니다!! UI 컴포넌트를 별도 패키지로 관리해 보면 어떨까요?

**Comment by hanjiwon1** [comment]
빌드 속도를 "엄청나게" 향상시킬 수 있다는걸 보니 안바꿀 이유가 없겠는데요 ㅎㅎ
UI 컴포넌트라면 버튼, 텍스트 같은 아토믹한 컴포넌트들인건가요~?

**Reply by rkskekzzz** [reply]
넵 맞습니다 ㅎㅎ

**Comment by ash0814** [comment]
저도 안바꿀 이유가 없어보입니다ㅎㅎ 별도 관리 좋아요!

**Comment by leekh716** [comment]
호우 조아유
버튼, 인풋같은 공통 UI말고 다른 컴포넌트UI들은 기존처럼 web/src/component 안에 있는거져??

**Reply by rkskekzzz** [reply]
넵 맞습니당!
